src/suites/cts/validation/createRenderPipeline.spec.ts

```diff
@@ -0,0 +1,233 @@
+
+  createTexture(params: { format: GPUTextureFormat; sampleCount: number }): GPUTexture {
+    const { format, sampleCount } = params;
+
+    return this.device.createTexture({
+      size: { width: 4, height: 4, depth: 1 },
+      usage: GPUTextureUsage.OUTPUT_ATTACHMENT,
+      format,
+      sampleCount,
+    });
+  }
+}
+
+export const g = new TestGroup(F);
+
+g.test('basic use of createRenderPipeline', t => {
+  const descriptor = t.getDescriptor();
+
+  t.device.createRenderPipeline(descriptor);
+});
+
+g.test('at least one color state is required', async t => {
+  const goodDescriptor = t.getDescriptor({
+    colorStates: [{ format: 'rgba8unorm' }],
+  });
+
+  // Control case
+  t.device.createRenderPipeline(goodDescriptor);
+
+  // Fail because lack of color states
+  const badDescriptor = t.getDescriptor({
+    colorStates: [],
+  });
+
+  await t.expectValidationError(() => {
+    t.device.createRenderPipeline(badDescriptor);
+  });
+});
+
+g.test('color formats must be renderable', async t => {
+  const goodDescriptor = t.getDescriptor({
+    colorStates: [{ format: 'rgba8unorm' }],
+  });
+
+  // Control case
+  t.device.createRenderPipeline(goodDescriptor);
+
+  // Fails because RG11B10Float is non-renderable
+  const badDescriptor = t.getDescriptor({
+    colorStates: [{ format: 'rg11b10float' }],
+  });
+
+  await t.expectValidationError(() => {
+    t.device.createRenderPipeline(badDescriptor);
+  });
+});
+
+g.test('sample count must be valid', async t => {
+  const goodDescriptor = t.getDescriptor({
+    sampleCount: 4,
+  });
+
+  // Control case
+  t.device.createRenderPipeline(goodDescriptor);
+
+  // Fails because 3 is not a power of 2.
+  const badDescriptor = t.getDescriptor({
+    sampleCount: 3,
+  });
+
+  await t.expectValidationError(() => {
+    t.device.createRenderPipeline(badDescriptor);
+  });
+});
+
+g.test('sample count must be equal to the one of every attachment in the render pass', async t => {
+  const { multisampledPass, multisampledPipeline, success } = t.params;
+
+  const sampleCountForTexture = multisampledPass ? 4 : 1;
+  const colorTexture = t.createTexture({
+    format: 'rgba8unorm',
+    sampleCount: sampleCountForTexture,
+  });
+  const depthStencilTexture = t.createTexture({
+    format: 'depth24plus-stencil8',
+    sampleCount: sampleCountForTexture,
+  });
+  const renderPassDescriptorWithoutDepthStencil = {
+    colorAttachments: [
+      {
+        attachment: colorTexture.createView(),
+        loadValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
+      },
+    ],
+  };
+  const renderPassDescriptorWithDepthStencilOnly = {
+    colorAttachments: [],
+    depthStencilAttachment: {
+      attachment: depthStencilTexture.createView(),
+      depthLoadValue: 1.0,
+      depthStoreOp: 'store',
+      stencilLoadValue: 0,
+      stencilStoreOp: 'store',
+    },
+  };
+
+  const sampleCountForPipeline = multisampledPipeline ? 4 : 1;
+  const pipelineWithoutDepthStencilDescriptor = t.getDescriptor({
```

Comment:
µnit: I'd just inline this variable below to avoid being too verbose (which makes it slightly harder to read)